server/src/main/java/io/druid/curator/CuratorModule.java

```diff
@@ -91,6 +104,58 @@ public void stop()
     return framework;
   }
 
+  @Provides
+  @LazySingleton
+  public EnsembleProvider makeEnsembleProvider(
+      CuratorConfig config, ExhibitorConfig exConfig, Lifecycle lifecycle
+  )
+  {
+    if (exConfig.getHosts().isEmpty()) {
+      return new FixedEnsembleProvider(config.getZkHosts());
+    }
+
+    final ExhibitorEnsembleProvider provider = new ExhibitorEnsembleProvider(
+        new Exhibitors(
+            exConfig.getHosts(),
+            exConfig.getRestPort(),
+            newBackupProvider(config.getZkHosts())
+        ),
+        new DefaultExhibitorRestClient(exConfig.getUseSsl()),
+        exConfig.getRestUriPath(),
+        exConfig.getPollingMs(),
+        new BoundedExponentialBackoffRetry(baseSleepTimeMs, maxSleepTimeMs, maxRetries)
+    );
+
+    lifecycle.addHandler(new Lifecycle.Handler()
+    {
+      @Override
+      public void start() throws Exception
+      {
+        log.info("Poll the list of zookeeper servers for initial ensemble");
+        provider.pollForInitialEnsemble();
```

Comment:
why this has to be done from lifecycle handler and not directly after creation of the provider ?